scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -344,12 +345,14 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
     thatNumKeys: Long,
     fpProb: Double = 0.01
   )(implicit hash: Hash128[K], koder: Coder[K], voder: Coder[V]): SCollection[(K, (V, W))] =
-    SCollection.unionAll(
-      splitSelfUsing(that, thatNumKeys, fpProb).map {
-        case (_, lhsOverlap, rhs) =>
-          lhsOverlap.join(rhs)
-      }
-    )
+    self.transform { me =>
+      SCollection.unionAll(
+        splitThisUsingThat(me, that, thatNumKeys, fpProb).map {
```

Comment:
I would probably just rename this to `split`

Reply:
Thanks for the feedback. It makes sense, I was a bit confused myself with the earlier name.
I'll make the change and push a new commit by tomorrow.